tap_s3_csv/s3.py

```diff
@@ -189,12 +410,12 @@ def get_input_files_for_table(config, table_spec, modified_since=None):
                 LOGGER.info("Found %s matching files and %s non-matching files",
                             matched_files_count, unmatched_files_count)
 
-    if 0 == matched_files_count:
+    if matched_files_count == 0:
         raise Exception("No files found matching pattern {}".format(pattern))
 
 
 @retry_pattern()
-def list_files_in_bucket(bucket, search_prefix=None):
+def list_files_in_bucket(bucket, search_prefix=None, recursive_search=True):
```

Comment:
For WP-7630

Reply:
i see in jira ticket this is also for discovery.  so to confirm

discovery = true
import = false

Reply:
I used True as the default because export also uses this tap, and I didn't want to break any existing functionality. I briefly looked and I don't think export depends on the recursive search functionality, but I wasn't 100% sure.

The "discovery" of all files in the bucket is done in main app `getAmazonS3Directory`, so for both discovery (getting column info for desired file) and import in the tap, `recursive_search` would be the same (both False, pending any new functionality requirements)